### PR TITLE
Feature/general overview

### PR DESCRIPTION
**Pull request by AurisAudentis**
This is a pull request merging the two largest parts of my thesis together. Therefore, it supersedes and includes both #660  and #718 . This does not include semantic matching.

Features:
- General overview starting page, with general information
- Expanded graph view with highlighting of clusters, nicer visualisations when selecting a node or cluster and more advanced visualisations for the clusters
- Individual plagiarism page with suggestions of which files might be interesting, with lots of context

### CODE REVIEW

File: cli/src/cli/views/webView.ts [hunk 1]
```diff
@@ -15,6 +15,6 @@ export class WebView extends FileView {
 
   async show(): Promise<void> {
     const reportDir = await this.writeToDirectory();
-    await runServer(reportDir, this.options);
+    setTimeout(async () => await runServer(reportDir, this.options), 1000);
```
[the remark this answers is not in the document]
Comment (AurisAudentis) [reply cli/src/cli/views/webView.ts:18]: Yes, on my system the files that are written to disk are not always finished when the server loads, leading to a false load without any files available. The 'await writeToDirectory' seems to not function entirely properly.

File: web/src/views/Cluster.vue [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+@Component({
+  components: { ClusteringTable }
+})
+export default class Cluster extends DataView {
+  created(): Promise<void> {
+    return super.ensureData();
+  }
+
+  get cluster(): Clustering {
+    return cluster(super.pairs, super.files, 0.28);
```
[the remark this answers is not in the document]
Comment (AurisAudentis) [reply web/src/views/Cluster.vue:26]: This entire file is unused, I will remove it.

File: web/vue.config.js [hunk 3]
```diff
@@ -19,6 +19,10 @@ module.exports = {
   },
   // explicitly makes sure that tree sitter is not loaded as it does not work in browser environments
   configureWebpack: {
+    devServer: {
+      host: "localhost",
+      headers: { "Access-Control-Allow-Origin": "*" }
```
[the remark this answers is not in the document]
Comment (AurisAudentis) [reply web/vue.config.js:24]: On my system, the local devserver tries to run on my public IP, which breaks the auto-reloading.

File: web/src/components/clustering/ClusteringFileTag.vue [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    super();
+    this.initialize();
+  }
+
+  async initialize(): Promise<void> {
+    await this.ensureData();
+    this.legend = this.createLegend();
+  }
+
+  getInitials(): string {
```
[the remark this answers is not in the document]
Comment (rien) [reply web/src/components/clustering/ClusteringFileTag.vue]: I see. This can stay for now and later we'll want to do some refactoring in the API models.

**Pull request merged**